scripts/migrateOldHfDataset.py

```diff
@@ -18,21 +18,30 @@
 from huggingface_hub import HfApi, login
 from dotenv import load_dotenv
 import sys
+import logging
 
 load_dotenv()
 
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
 def dlData(year):
     '''
     Download the dataset from Edwin's huggingface
     '''
     url = f"https://huggingface.co/datasets/edwinjue/311-data-{year}/resolve/main/{year}.csv"
     outfile = f"{year}.csv"
+    chunk_size = 1024 * 1024 # 1 MB
+
     response = requests.get(url, stream=True)
 
     # Save downloaded file
     with open(outfile, "wb") as file:
-        for data in tqdm(response.iter_content()):
-            file.write(data)
+        for chunk in tqdm(response.iter_content(chunk_size=chunk_size), desc="Downloading data"):
+            if chunk:  # filter out keep-alive new chunks
```

Comment:
were you encountering keep-alive chunks while using this script? just curious how this was affecting the result csv

Reply:
this is to handle HTTP's keep-alive mechanism which may send empty chunks to keep the connection open especially for downloading larger files. i didnt check to see if there were empty chunks downloaded prior, but just implement check if the current chunk is not empty as a precaution, so that it only downloads non-empty chunks.